### PR TITLE
plugins/codeium-nvim: init

### LINKED ISSUE #825: [PLUGIN REQUEST] `codeium.nvim`
**Plugin**: `codeium.nvim`
**Plugin repository**: https://github.com/Exafunction/codeium.nvim
**Is the plugin on nixpkgs**: https://github.com/NixOS/nixpkgs/pull/276080

Related: https://github.com/nix-community/nixvim/issues/824

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [codeium.nvim](https://github.com/Exafunction/codeium.nvim).

Fixes #825

### CODE REVIEW

File: plugins/completion/nvim-cmp/sources/codeium-nvim.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+in {
+  meta.maintainers = [maintainers.GaetanLepage];
+
+  options.plugins.codeium-nvim =
+    helpers.extraOptionsOptions
+    // {
+      package = helpers.mkPackageOption "codeium.nvim" pkgs.vimPlugins.codeium-nvim;
+
+      configPath =
+        helpers.defaultNullOpts.mkNullable (with types; either str helpers.rawType)
```
Comment (traxys) [inline plugins/completion/nvim-cmp/sources/codeium-nvim.nix:19]: You should now be able to simply use `mkStr`, it should accept raw types

File: plugins/completion/nvim-cmp/sources/codeium-nvim.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        ''{__raw = "vim.fn.stdpath('cache') .. '/codeium/config.json'";}''
+        "The path to the config file, used to store the API key.";
+
+      binPath =
+        helpers.defaultNullOpts.mkNullable (with types; either str helpers.rawType)
```
Comment (traxys) [inline plugins/completion/nvim-cmp/sources/codeium-nvim.nix:24]: Same

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/completion/nvim-cmp/sources/codeium-nvim.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/completion/nvim-cmp/sources/codeium-nvim.nix]: Done !

File: plugins/completion/nvim-cmp/sources/codeium-nvim.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/completion/nvim-cmp/sources/codeium-nvim.nix]: Done

**Review by traxys — APPROVED** [review]

**Pull request merged**